Source/Session.swift

```diff
@@ -255,81 +265,98 @@ open class Session {
     open func upload(_ data: Data,
                      to convertible: URLConvertible,
                      method: HTTPMethod = .post,
-                     headers: HTTPHeaders? = nil) -> UploadRequest {
+                     headers: HTTPHeaders? = nil,
+                     interceptor: RequestInterceptor? = nil) -> UploadRequest {
         let convertible = ParameterlessRequestConvertible(url: convertible, method: method, headers: headers)
 
-        return upload(data, with: convertible)
+        return upload(data, with: convertible, interceptor: interceptor)
     }
 
-    open func upload(_ data: Data, with convertible: URLRequestConvertible) -> UploadRequest {
-        return upload(.data(data), with: convertible)
+    open func upload(_ data: Data,
+                     with convertible: URLRequestConvertible,
+                     interceptor: RequestInterceptor? = nil) -> UploadRequest {
```

Comment:
More struggling here with the external API. Some methods are more verbose than others, so trying to make some read nicely and others not is challenging.

Reply:
Swift guidelines say that defaulted arguments should come after non-defaulted ones, so this looks good.